pkg/solana/chain_test.go

```diff
@@ -371,3 +318,305 @@ func TestChain_Transact(t *testing.T) {
+		require.NoError(t, err)
+		require.NotNil(t, result)
+		require.NotNil(t, *result)
+		require.Error(t, (*result).TxError())
+		require.Equal(t, mn.Fatal, (*result).Code())
+		require.Empty(t, (*result).Signature())
+	})
+}
+
+func TestSolanaChain_MultiNode_Txm(t *testing.T) {
+	cfg := solcfg.NewDefault()
+	cfg.MultiNode.MultiNode.Enabled = ptr(true)
+	cfg.Nodes = []*solcfg.Node{
+		{
+			Name: ptr("primary-1"),
+			URL:  config.MustParseURL(client.SetupLocalSolNode(t)),
+		},
+		{
+			Name: ptr("primary-2"),
+			URL:  config.MustParseURL(client.SetupLocalSolNode(t)),
+		},
+		{
+			Name:     ptr("sendonly-1"),
+			URL:      config.MustParseURL(client.SetupLocalSolNode(t)),
+			SendOnly: true,
+		},
+		{
+			Name:     ptr("sendonly-2"),
+			URL:      config.MustParseURL(client.SetupLocalSolNode(t)),
+			SendOnly: true,
+		},
+	}
+
+	// setup keys
+	key, err := solana.NewRandomPrivateKey()
+	require.NoError(t, err)
+	pubKey := key.PublicKey()
+
+	// setup receiver key
+	privKeyReceiver, err := solana.NewRandomPrivateKey()
+	require.NoError(t, err)
+	pubKeyReceiver := privKeyReceiver.PublicKey()
+
+	// mocked keystore
+	mkey := mocks.NewSimpleKeystore(t)
+	mkey.On("Sign", mock.Anything, pubKey.String(), mock.Anything).Return(func(_ context.Context, _ string, data []byte) []byte {
+		sig, _ := key.Sign(data)
+		return sig[:]
+	}, nil)
+	mkey.On("Sign", mock.Anything, pubKeyReceiver.String(), mock.Anything).Return([]byte{}, config.KeyNotFoundError{ID: pubKeyReceiver.String(), KeyType: "Solana"})
+
+	testChain, err := newChain("localnet", cfg, mkey, logger.Test(t))
+	require.NoError(t, err)
+
+	err = testChain.Start(tests.Context(t))
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, testChain.Close())
+	}()
+
+	// fund keys
+	client.FundTestAccounts(t, []solana.PublicKey{pubKey}, cfg.Nodes[0].URL.String())
+
+	// track initial balance
+	selectedClient, err := testChain.getClient()
+	require.NoError(t, err)
+	receiverBal, err := selectedClient.Balance(tests.Context(t), pubKeyReceiver)
+	assert.NoError(t, err)
+	assert.Equal(t, uint64(0), receiverBal)
+
+	createTx := func(signer solana.PublicKey, sender solana.PublicKey, receiver solana.PublicKey, amt uint64) *solana.Transaction {
+		selectedClient, err = testChain.getClient()
+		assert.NoError(t, err)
+		hash, hashErr := selectedClient.LatestBlockhash(tests.Context(t))
+		assert.NoError(t, hashErr)
+		tx, txErr := solana.NewTransaction(
+			[]solana.Instruction{
+				system.NewTransferInstruction(
+					amt,
+					sender,
+					receiver,
+				).Build(),
+			},
+			hash.Value.Blockhash,
+			solana.TransactionPayer(signer),
+		)
+		require.NoError(t, txErr)
+		return tx
+	}
+
+	// Send funds twice, along with an invalid transaction
+	require.NoError(t, testChain.txm.Enqueue(tests.Context(t), "test_success", createTx(pubKey, pubKey, pubKeyReceiver, solana.LAMPORTS_PER_SOL)))
+	time.Sleep(500 * time.Millisecond) // pause 0.5s for new blockhash
```

Comment:
What happens if block production is a bit slower?

Reply:
Since this is using the **Solana Test Validator** it should always work as expected. The same approach is being used in the current txm tests.

Reply:
If the test fails on failure to produce a new block within 0.5s we should avoid using `time.Sleep` and instead explicitly wait for a new block to be produced. 
Otherwise, we might experience flaky behavior due to limited resources when the whole test suite runs in the CI.